### PR TITLE
Feat/dyno routes

### LINKED ISSUE #23: Vue Dynamic Routing
Find a way to display dynamic routes for each single campaign

### PR DESCRIPTION
**Pull request by edmdc**
# Pull request checklist

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Current behavior

- No Dynamic routes in place

## New behavior

- Campaign details are dynamically displayed using route params

## Other information

Closes #23 

image
**Note the URL change**
image

### CODE REVIEW

**Comment by edmdc** [comment]
Please select rebase and merge from the Merge Pull Request Drop down menu.

File: src/components/CampaignCard.vue [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <div>
     <section class="campaign-card">
       <h1> {{campaign.title}}</h1>
-      <router-link to="/campaigns/0948dx83ds">
+      <router-link :to="{ name: 'Single Campaign', params: {address: campaign.address} }">
```
Comment (edmdc) [inline src/components/CampaignCard.vue:5]: Create new link on card and use campaign address as dynamic route path

File: src/components/CampaignDetails.vue [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+export default {
+  name: 'CampaignDetails',
+  props: ['address'],
+  computed: {
+    campaign() {
+      return this.$store.getters.getSingleCampaign(Number(this.address))
+    }
+  }
+}
+</script>
```
Comment (edmdc) [inline src/components/CampaignDetails.vue:24]: Gets address from url, and details from store

File: src/router/index.js [hunk 3]
```diff
@@ -1,24 +1,25 @@
...
-    path: '/campaigns/:address',
-    name: 'Single Campaign',
+    path: "/campaigns/:address",
+    name: "Single Campaign",
     // route level code-splitting
     // this generates a separate chunk (about.[hash].js) for this route
     // which is lazy-loaded when the route is visited.
-    component: () => import('../views/campaigns/singleCampaign.vue')
+    component: () => import("../components/CampaignDetails.vue"),
+    props: (route) => ({ ...route.params })
```
Comment (edmdc) [inline src/router/index.js:17]: Dynamic creation & attachment of props (address)

**Review by attom2 — APPROVED** [review]

File: src/components/CampaignCard.vue (hunk 1 above)
Comment (attom2) [reply src/components/CampaignCard.vue:5]: Nice I like this

**Review by jpc20 — APPROVED** [review]
Looks great! Soon this will be actual addresses on the blockchain.

File: src/components/CampaignDetails.vue (hunk 2 above)
Comment (attom2) [reply src/components/CampaignDetails.vue:24]: Very semantic with the getters

**Pull request merged**